sktime/classification/model_evaluation/_functions.py

```diff
@@ -0,0 +1,481 @@
+            warnings.warn(
+                f"""
+                In evaluate, fitting of classifier {type(classifier).__name__} failed,
+                you can set error_score='raise' in evaluate to see
+                the exception message.
+                Fit failed for the {i}-th data split, on training data y_train
+                , and len(y_train)={len(y_train)}.
+                The score will be set to {error_score}.
+                Failed classifier with parameters: {classifier}.
+                """,
+                FitFailedWarning,
+                stacklevel=2,
+            )
+
+    # Storing the remaining evaluate detail
+    temp_result["fit_time"] = [fit_time]
+
+    if return_data:
+        temp_result["X_train"] = [X_train]
+        temp_result["X_test"] = [X_test]
+        temp_result["y_train"] = [y_train]
+        temp_result["y_test"] = [y_test]
+        temp_result.update(y_preds_cache)
+    result = pd.DataFrame(temp_result)
+    column_order = _get_column_order_and_datatype(scoring, return_data)
+    result = result.reindex(columns=column_order.keys())
+
+    return result, classifier
+
+
+def evaluate(
+    classifier,
+    cv=KFold(n_splits=3, shuffle=False),
+    X=None,
+    y=None,
+    scoring: Optional[Union[callable, list[callable]]] = None,
+    return_data: bool = False,
+    error_score: Union[str, int, float] = np.nan,
+    backend: Optional[str] = None,
+    backend_params: Optional[dict] = None,
+):
+    r"""Evaluate classifier using timeseries cross-validation.
+
+    All-in-one statistical performance benchmarking utility for classifiers
+    which runs a simple backtest experiment and returns a summary pd.DataFrame.
+
+    The experiment run is the following:
+
+    Train and test folds are generated by ``cv.split(X, y)``.
+    For each fold ``i`` in ``K`` number of folds:
+    1. Fit the ``classifier`` to the training set ``S_i``
+    2. Use the ``classifier`` to make a prediction ``y_pred`` on the  test set ``T_i``
+    3. Compute the score using ``scoring(T_i, y_pred)``
+    4. If ``i == K``, terminate, otherwise
+    5. Set ``i = i + 1`` and go to 1
+
+    Results returned in this function's return are:
+
+    * results of ``scoring`` calculations, from 4,  in the ``i``-th loop
+    * runtimes for fitting and/or predicting, from 2, 3, in the ``i``-th loop
+    * :math:`y_{train, i}`, :math:`y_{test, i}`, ``y_pred`` (optional)
+
+    A distributed and-or parallel back-end can be chosen via the ``backend`` parameter.
+
+    Parameters
+    ----------
+    classifier : sktime BaseClassifier descendant (concrete classifier)
+         sktime classifier to benchmark
+    cv : sklearn KFold
+        Provides train/test indices to split data in train/test sets.
+       Splits the dataset into k consecutive folds (without shuffling by default).
+    X : sktime compatible time series panel data container, Panel scitype, e.g.,
+             pd-multiindex: pd.DataFrame with columns = variables,
```

Comment:
this formatting will not properly render - can you look for a docstring where this is a bullet point list and renders properly?